pkg/daemon/ceph/cleanup/radosnamespace.go

```diff
@@ -91,3 +102,29 @@ func BlockPoolCleanup(context *clusterd.Context, clusterInfo *cephclient.Cluster
 	logger.Infof("successfully cleaned up CephBlockPool %q resource", poolName)
 	return nil
 }
+
+func AddClientIPsToBlocklist(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, images []cephclient.CephBlockImage, poolName, radosNamespace string) error {
+	for _, image := range images {
+		rbdStatus, err := cephclient.GetRBDImageStatus(context, clusterInfo, poolName, image.Name, radosNamespace)
+		if err != nil {
+			return errors.Wrapf(err, "failed to list watchers for the image %q in %s", image.Name, radosNamespace)
+		}
+		imageWatchers := rbdStatus.GetWatcherIPs()
+		if len(imageWatchers) == 0 {
+			logger.Info("no watchers found for image %q in pool %q in namespace %q", image.Name, poolName, radosNamespace)
+			return nil
+		}
+
+		logger.Infof("watchers for image %q in pool %q in namespace %q: %v", image.Name, poolName, radosNamespace, imageWatchers)
+		for _, watcher := range imageWatchers {
+			logger.Infof("blocklist watcher %q for image %q in pool %q in namespace %q", watcher, image.Name, poolName, radosNamespace)
+			err := cephclient.BlocklistIP(context, clusterInfo, watcher, ClientBlocklistDuration)
```

Comment:
If there are multiple watchers for the same IP, we only want to blocklist each of them once. So we likely need a map to keep track of which ones have been blocklisted so we can skip any duplicates.